src/posit/connect/bundles.py

```diff
@@ -99,14 +96,66 @@ def delete(self) -> None:
         url = urls.append(self.config.url, path)
         self.session.delete(url)
 
+    def download(self) -> bytes:
+        """Download a bundle.
+
+        Returns
+        -------
+        bytes
+            Archive contents in bytes representation.
```

Comment:
Will there ever be a use for this? Won't you always want to write this to a file?

Reply:
We can inspect in memory, but I'm not sure if that will be useful. @mmarchetti, what are your thoughts? Do we need to provide support for inspecting the tarball in memory?

Reply:
I think a download to file option is probably sufficient. If we do add an in-memory option, it would be better to return a BytesIO rather than to read the contents immediately.